projects/dotcms-field-elements/src/components/dot-date-range/dot-date-range.e2e.ts

```diff
@@ -1,147 +1,349 @@
-import { E2EElement, E2EPage, newE2EPage } from '@stencil/core/testing';
+import { newE2EPage, E2EElement, E2EPage } from '@stencil/core/testing';
 import { EventSpy } from '@stencil/core/dist/declarations';
+import { dotTestUtil } from '../../utils';
+
+const getDays = (page: E2EPage) => page.findAll('.flatpickr-day');
 
 describe('dot-date-range', () => {
     let page: E2EPage;
     let element: E2EElement;
     let input: E2EElement;
     let spyStatusChangeEvent: EventSpy;
-    let spyValueChange: EventSpy;
-
-    beforeEach(async () => {
-        page = await newE2EPage({
-            html: `
-              <dot-date-range
-                    label='Name:'
-                    min="2019-01-01"
-                    max="2019-12-31"
-                    name='dateRange'
-                    value='2019-5-1,2019-5-8'
-                    hint='this is a hint'
-                    required=true
-                    required-message="Required Name"
-                ></dot-date-range>`
-        });
-
-        spyStatusChangeEvent = await page.spyOnEvent('statusChange');
-        spyValueChange = await page.spyOnEvent('valueChange');
-        element = await page.find('dot-date-range');
-        input = await page.find('input');
-    });
+    let spyValueChangeEvent: EventSpy;
 
-    it('should render', () => {
-        // tslint:disable-next-line:max-line-length
-        const tagsRenderExpected = `<dot-date-range label=\"Name:\" min=\"2019-01-01\" max=\"2019-12-31\" name=\"dateRange\" value=\"2019-5-1,2019-5-8\" hint=\"this is a hint\" required=\"true\" required-message=\"Required Name\" class=\"dot-valid dot-required hydrated dot-dirty dot-touched\"><div class=\"dot-field__label\"><label for=\"dot-dateRange\">Name:</label><span class=\"dot-field__required-mark\">*</span></div><input class=\"flatpickr-input\" id=\"dateRange\" required=\"\" type=\"text\" readonly=\"readonly\"><select><option value=\"0\">Date Presets</option><option value=\"-7\">Last Week</option><option value=\"7\">Next Week</option><option value=\"-30\">Last Month</option><option value=\"30\">Next Month</option></select><span class=\"dot-field__hint\">this is a hint</span></dot-date-range>`;
-        expect(element.outerHTML).toBe(tagsRenderExpected);
-    });
+    describe('render', () => {
+        beforeEach(async () => {
+            page = await newE2EPage();
+        });
 
-    it('should set default value', async() => {
-        const dateValue = await input.getProperty('value');
-        expect(dateValue).toBe('2019-05-01,2019-05-08');
-    });
+        describe('CSS classes', () => {
```

Comment:
Doesn't apply the 6 class states we have?

Reply:
no it doesn't